### PR TITLE
Added 1 second timeout to read data in readLine

### PR DESCRIPTION
**Pull request by ONLYstcm**
A simple 1 second connection timeout limit that will abort a connection request if a response is not received in time to address [Issue 78](https://github.com/gilmaimon/ArduinoWebsockets/issues/78). Tested with ESP32

### CODE REVIEW

**Review by 3imed-jaberi — APPROVED** [review]
+1

File: src/tiny_websockets/network/generic_esp/generic_esp_clients.hpp [hunk 1]
```diff
@@ -3,6 +3,9 @@
 #include <tiny_websockets/internals/ws_common.hpp>
 #include <tiny_websockets/network/tcp_client.hpp>
 
+// Timeout for connection attempts in milliseconds
+#define readLineTimeout 1000
```
Comment (gilmaimon) [inline src/tiny_websockets/network/generic_esp/generic_esp_clients.hpp:7]: Keep the CONSTANTS_IN_CAPITAL_CASE convention 
Also, put it in [ws_config_defs.hpp](https://github.com/gilmaimon/ArduinoWebsockets/blob/master/src/tiny_websockets/ws_config_defs.hpp)

File: src/tiny_websockets/network/generic_esp/generic_esp_clients.hpp [hunk 2]
```diff
@@ -51,7 +54,9 @@ namespace websockets { namespace network {
       WSString line = "";
 
       int ch = -1;
+      uint64_t _timer = millis();
```
Comment (gilmaimon) [inline src/tiny_websockets/network/generic_esp/generic_esp_clients.hpp:57]: No need for the leading underscore (see other variable names..)

Also consider using a more indicative name, like `millisBeforeReadingHeaders`

File: src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp [hunk 3]
```diff
@@ -6,6 +6,9 @@
 #include <tiny_websockets/network/tcp_client.hpp>
 #include <tiny_websockets/network/tcp_server.hpp>
 
+// Timeout for connection attempts in milliseconds
+#define readLineTimeout 1000
```
Comment (gilmaimon) [inline src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp:10]: Use the one from config_defs (or find a better common place for this :))

**Review by gilmaimon — CHANGES_REQUESTED** [review]
Hi, sorry for the delay.

I posted some review comments on the code, once those will be refactored I'll merge :)

**Comment by ONLYstcm** [comment]
@gilmaimon please check the new changes as per your review :)

File: src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp [hunk 4]
```diff
@@ -59,8 +59,12 @@ namespace websockets { namespace network {
       WSString line = "";
 
       int ch = -1;
+
+      uint64_t millisBeforeReadingHeaders = millis();
       while( ch != '\n' && available()) {
+        if ( millis() - _timer > readLineTimeout) return "";
```
Comment (gilmaimon) [inline src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp:65]: I assume you need to remove this line?

File: src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp [hunk 5]
```diff
@@ -59,8 +59,12 @@ namespace websockets { namespace network {
       WSString line = "";
 
       int ch = -1;
+
+      uint64_t millisBeforeReadingHeaders = millis();
```
Comment (gilmaimon) [inline src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp:63]: `const uint64_t millisBeforeReadingHeaders`

File: src/tiny_websockets/network/generic_esp/generic_esp_clients.hpp [hunk 6]
```diff
@@ -51,7 +51,10 @@ namespace websockets { namespace network {
       WSString line = "";
 
       int ch = -1;
+
+      uint64_t millisBeforeReadingHeaders = millis();
```
Comment (gilmaimon) [inline src/tiny_websockets/network/generic_esp/generic_esp_clients.hpp:55]: `const uint64_t millisBeforeReadingHeaders`

**Review by gilmaimon — CHANGES_REQUESTED** [review]
@ONLYstcm Here are some more tiny comments. I think you forgot to remove on of the lines that use `_timer` so that file will probably not compile.. Also, see 2 tiny `const` comments.

Thanks!

File: src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp (hunk 4 above)
Comment (ONLYstcm) [reply src/tiny_websockets/network/teensy41/teensy41_tcp_client.hpp:65]: My apologies I pushed the wrong commit, this was supposed to be removed

**Comment by ONLYstcm** [comment]
2nd pass of reviews added

**Comment by gilmaimon** [comment]
Thanks, merged!

**Pull request merged**